conda_forge_tick/utils.py

```diff
@@ -517,30 +527,145 @@ def parse_recipe_yaml_local(
         for some errors. Have fun.
     """
 
-    return _render_recipe_yaml(text)[0]["recipe"]
+    rendered_recipe = _render_recipe_yaml(text)[0]["recipe"]
 
 
-def _render_recipe_yaml(text: str) -> dict[str, Any]:
+def _render_recipe_yaml(
+    text: str,
+    cbc_path=None,
+) -> list[dict[str, Any]]:
     """
     Renders the given recipe YAML text using the `rattler-build` command-line tool.
 
-    Args:
-        text (str): The recipe YAML text to render.
+    Parameters
+    ----------
+    text : str
+        The recipe YAML text to render.
+    cbc_path : str, optional
+        The path to global pinning file.
 
-    Returns:
-        dict[str, Any]: The rendered recipe as a dictionary.
+    Returns
+    -------
+    dict[str, Any]
+        The rendered recipe as a dictionary.
     """
+    variant_config_flags = [] if cbc_path is None else ["--variant-config", cbc_path]
     res = subprocess.run(
-        ["rattler-build", "build", "--render-only"],
+        ["rattler-build", "build", "--render-only"] + variant_config_flags,
         stdout=subprocess.PIPE,
-        stderr=subprocess.DEVNULL,
         text=True,
         input=text,
         check=True,
     )
     return json.loads(res.stdout)
 
 
+def _validate_parsed_recipes(
```

Comment:
One thing to be aware of here is that conda-forge has tons of invalid recipe bits. I don't see this changing even with the new format. So we cannot run validation that would make the bot error during its processing. We also need the code below to be robust to invalid recipes. It can mark the node as a parsing error if we want, but it needs to continue working. IDK what the code does right now, but I did want to make the comment before we got much further.

Reply:
Thanks for giving the feedback early. I was hoping that we could error out when we encounter invalid `recipe.yaml`. If that's not an option, I will adapt the code.

> It can mark the node as a parsing error if we want, but it needs to continue working

Is there already something for that in place.

Reply:
Yes the way we handle errors in downloads of the repos.